routes/authors.js

```diff
@@ -1,8 +1,55 @@
 const express = require("express");
 const generateId = require("uuid/v4");
+const authors = require("../data/authors");
+const body = require('body-parser')
+const books = require("../data/books");
+
+
 
 const router = express.Router();
 
 // Define your endpoints here
 
-module.exports = router;
+router.get('/', (req, res, next) => {
+	const sorter = req.query.sort
+	if (sorter) res.json(authors.map(el => el[sorter]).sort())
+	res.json(authors)
+})
+
+router.get('/=', (req, res, next) => {
+	res.json(authors.map(el => el.name))
+})
+
+
+router.get('/:id', (req, res, next) => {
+	res.json(authors.filter(el => el.id === req.params.id))
```

Comment:
Good idea. However, filter will return an array, whereas you only want to send an object. You would use the `find` method to send only the first match instead.

By the way, you also need to handle the case where an author doesn't exist.

Reply:
Yeah, I ran into some trouble with that in the patch route, forgot to apply the [0] to the other ones